src/main/java/core/basesyntax/service/DataHandler.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.model.Fruit;
+
+public interface DataHandler {
+    void processData(StorageDao storageDao, Fruit fruit, int quantity);
+}

```

Comment:
lets fix naming DataHandler -> OperationHandler

and you should not pass dao and a parameter, just create this as variable inside each implementation